tests/test-runner.cc

```diff
@@ -77,11 +77,23 @@ compartment_error_handler(struct ErrorState *frame, size_t mcause, size_t mtval)
 	return ErrorRecoveryBehaviour::InstallContext;
 }
 
+__attribute__((noinline, weak)) void *pcc_as_sentry()
+{
+	return __builtin_return_address(0);
+}
+
 /**
  * Test suite entry point.  Runs all of the tests that we have defined.
  */
+[[cheriot::interrupt_state(disabled)]]
 int __cheri_compartment("test_runner") run_tests()
 {
+	// This is started as an interrupts-disabled thread, make sure that it really is!
+	Capability sealedPCC = pcc_as_sentry();
+	TEST((sealedPCC.type() == CheriSealTypeReturnSentryDisabling) ||
+	       (sealedPCC.type() == CheriSealTypeSentryDisabling),
```

Comment:
Shouldn't it always be a return sentry? Why do you need the `||`?

Reply:
Sonata 0.2